### PR TITLE
Create guide for Squash/Merge

### PR DESCRIPTION
**Pull request by CaitlinV39**
## Description
Create a markdown that describes the squash merge requirements. Following these requirements will help us deliver release notes.

Also remove Data Migration from SchemaVersioning document.

### CODE REVIEW

File: SquashMergeRequirements.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Squash/Merge Requirements
+
+When commiting your PR, please make sure to do these steps at Squash/Merge to assist with Release Note creation.
+
+1. **Update the title** of your PR to be descriptive and less than 68 characters
```
[the remark this answers is not in the document]
Comment (CaitlinV39) [reply SquashMergeRequirements.md]: Updated!

**Comment by CaitlinV39** [comment]
@brendankowitz is there anything else needed before we merge?

**Review by brendankowitz — APPROVED** [review]

**Pull request merged**